modules/core/src/test/scala/org/tessellation/infrastructure/trust/TrustModelSuite.scala

```diff
@@ -0,0 +1,37 @@
+package org.tessellation.infrastructure.trust
+
+import cats.syntax.applicative._
+
+import org.tessellation.infrastructure.trust.TrustModel.calculateTrust
+import org.tessellation.infrastructure.trust.generators.{genPeerPublicTrust, genPeerTrustInfo}
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.sdk.domain.trust.storage.{PublicTrustMap, TrustMap}
+
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustModelSuite extends SimpleIOSuite with Checkers {
+  test("peer labels contain undiscovered peers does not throw an exception") {
+    val gen = for {
+      trust <- Gen.mapOfN(4, genPeerTrustInfo)
+      labels <- Gen.mapOfN(4, genPeerPublicTrust)
+      publicTrustMap = PublicTrustMap(labels)
+      peerId <- peerIdGen
+    } yield (TrustMap(trust, publicTrustMap), peerId)
+
+    forall(gen) {
+      case (trust, selfId) =>
+        for {
+          trustKeySet <- trust.trust.keySet.pure
+          labelKeySet = trust.peerLabels.value.keySet
+          result <- calculateTrust(trust, selfId).pure.attempt // expect to not throw an exception
+        } yield
+          expect.all(
+            labelKeySet.diff(trustKeySet).nonEmpty,
+            result.isRight
```

Comment:
you have an explicit check so I think the commend is not needed 🤔

Reply:
The comment has been removed.  The `labelKeySet.diff(trustKeySet).nonEmpty` is there because the exception is only thrown when the `TrustMap`'s `publicTrustMap` has `PeerId`s that `TrustMap`'s `trust` does not. That `expect` is there to make sure that is the case.

Reply:
So the error is thrown or isn't thrown? The test checks if it doesn't throw. I'm a little bit confused

Reply:
The error is not thrown now. The test was added because, after the rework of the `TrustStorage`, the `TrustModel` started throwing exceptions under certain conditions. It is desired to re-factor `TrustModel` to improve clarity. However, that is going to be done in a separate ticket. In the meantime, the existing `TrustModel` was patched and this test was added to ensure that the `TrustModel` won't throw an exception in this scenario.

Reply:
:+1: